nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -93,6 +99,16 @@ def __init__(
         self.virtual_tokens = virtual_tokens
         self.tokens_to_generate = tokens_to_generate
         self.truncation_method = truncation_method
+        if special_tokens is None:
+            self.special_tokens = {
+                "system_turn_start": "<extra_id_0>",
+                "turn_start": "<extra_id_1>",
+                "label_start": "<extra_id_2>",
+                "end_of_turn": "\n",
+                "end_of_name": "\n",
+            }
+        else:
+            self.special_tokens = special_tokens
 
```

Comment:
can we do a check to see if the tokens in special_tokens are tokenizer's special tokens or not? If not (the case with llama), can we just throw a warning that we'll use text as turn tokens which _might_ cause incorrect merging

Reply:
that is different, the token merge can still happen during multi-turn

what I mean is that if the turn tokens are not special tokens, we just say that there might be an error possible

Reply:
The `header_len` stops at the "end_of_turn". The next token is "turn_start". If the merge happens this assert will catch it. The multiple turn has the same thing. each turn ends with "end_of_turn" and the next token is "turn_start". So this one is enough to catch it. 

Also I don't see the point of just giving a warning which doesn't help the user at all.